Bottom: 70fc155cccbd4ada8765023f0bf426c20c01f3e5
Top:    ae4543df28a500661732e3345c74d668c358dfcf
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 15:09:01 +0800

autofs-5.0.7 - fix possible use after free in lookup_dir.c:lookup_init()

Add a missing error return in lookup_dir.c:lookup_init().


---

diff --git a/modules/lookup_dir.c b/modules/lookup_dir.c
index 07471b7..cbeda1f 100644
--- a/modules/lookup_dir.c
+++ b/modules/lookup_dir.c
@@ -98,6 +98,7 @@ int lookup_init(const char *mapfmt, int argc, const char *const *argv, void **co
 		free(ctxt);
 		warn(LOGOPT_NONE, MODPREFIX
 		     "dir map %s, is not a directory", argv[0]);
+		return 1;
 	}
 
 	*context = ctxt;
